sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala

```diff
@@ -712,7 +712,9 @@ abstract class SparkStrategies extends QueryPlanner[SparkPlan] {
       case r: logical.Range =>
         execution.RangeExec(r) :: Nil
       case r: logical.RepartitionByExpression =>
-        val shuffleOrigin = if (r.optNumPartitions.isEmpty) {
+        val shuffleOrigin = if (r.partitionExpressions.isEmpty && r.optNumPartitions.isEmpty) {
+          REPARTITION_BY_NONE
+        } else if (r.optNumPartitions.isEmpty) {
           REPARTITION
```

Comment:
Let's fully adopt the naming suggestion: REPARTITION_BY_COL, REPARTITION_BY_NUM, REPARTITION_BY_COL_AND_NUM, REPARTITION_BY_NONE

Reply:
@JkSelf I think we don't need REPARTITION_BY_COL_AND_NUM, because REPARTITION_BY_NUM already indicates no optimization can be done.

Reply:
OK